e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/generate-keying-sheet.spec.js

```diff
@@ -0,0 +1,146 @@
+  };
+  const PREVIOUS_PERIOD_FIXED_FEE = 1000;
+  const PREVIOUS_PERIOD_UTILISATION = 200000;
+  const FACILITY_UTILISATION_DATA = FacilityUtilisationDataEntityMockBuilder.forId('11111111')
+    .withReportPeriod(PREVIOUS_REPORT_PERIOD)
+    .withUtilisation(PREVIOUS_PERIOD_UTILISATION)
+    .withFixedFee(PREVIOUS_PERIOD_FIXED_FEE)
+    .build();
+
+  const ONE_YEAR_AGO = subYears(TODAY, 1);
+  const ONE_YEAR_FROM_CURRENT_REPORT_PERIOD_START = addYears(START_OF_CURRENT_REPORT_PERIOD, 1); // days left in cover period = 365
+
+  const FIRST_FEE_RECORD_ID = '11';
+  const firstFeeRecordBuilder = FeeRecordEntityMockBuilder.forReport(CURRENT_PERIOD_UTILISATION_REPORT)
+    .withId(FIRST_FEE_RECORD_ID)
+    .withStatus(FEE_RECORD_STATUS.TO_DO)
+    .withFacilityUtilisationData(FACILITY_UTILISATION_DATA)
+    .withBaseCurrency(CURRENCY.GBP)
+    .withFacilityUtilisation(CURRENT_PERIOD_UTILISATION)
+    .withPaymentCurrency(CURRENCY.GBP)
+    .withFeesPaidToUkefForThePeriodCurrency(CURRENCY.GBP)
+    .withFeesPaidToUkefForThePeriod(100);
+
+  const SECOND_FEE_RECORD_ID = '22';
+  const secondFeeRecordBuilder = FeeRecordEntityMockBuilder.forReport(CURRENT_PERIOD_UTILISATION_REPORT)
+    .withId(SECOND_FEE_RECORD_ID)
+    .withFacilityUtilisationData(FACILITY_UTILISATION_DATA)
+    .withBaseCurrency(CURRENCY.GBP)
+    .withFacilityUtilisation(CURRENT_PERIOD_UTILISATION)
+    .withPaymentCurrency(CURRENCY.GBP)
+    .withFeesPaidToUkefForThePeriodCurrency(CURRENCY.GBP)
+    .withFeesPaidToUkefForThePeriod(100);
+
+  const TFM_FACILITY = {
+    facilitySnapshot: {
+      coverStartDate: ONE_YEAR_AGO.getTime(),
+      coverEndDate: ONE_YEAR_FROM_CURRENT_REPORT_PERIOD_START.getTime(), // days left in cover period = 365
+      interestPercentage: 5,
+      dayCountBasis: 365,
+      ukefFacilityId: FACILITY_UTILISATION_DATA.id,
+    },
+  };
+
+  beforeEach(() => {
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, [TFM_FACILITY]);
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+
+    cy.task(NODE_TASKS.GET_ALL_BANKS).then((banks) => {
+      const bankWithReportToReconcile = banks.find((bank) => bank.id === BANK_ID);
+      cy.wrap(bankWithReportToReconcile).should('not.be.undefined');
+    });
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [CURRENT_PERIOD_UTILISATION_REPORT]);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+  });
+
+  it('can generate keying data when only one fee record for a facility is at MATCH status but does not generate the fixed fee adjustment or principal balance adjustment', () => {
+    const matchingPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withAmount(100).build();
+    const firstFeeRecord = firstFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.MATCH).withPayments([matchingPayment]).build();
+    const secondFeeRecord = secondFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.TO_DO).build();
```

Comment:
I'm not clear on how reusing the same builder across multiple tests is actually working since we are calling build multiple times now? 
Also, is this not a risk of inter test dependency, in that if you were to call `.withBaseCurrency(CURRENCY.EUR)` here would the builder have the value `CURRENCY.GBP` or `CURRENCY.EUR` at the start of the next test?

Reply:
I didn't actually think of that - my brain was in a functional world when I was writing this so I forgot about the side effect issues. Will re-work this